res/controllers/Reloop-Ready-scripts.js

```diff
@@ -0,0 +1,1141 @@
+            midi: [0xBE, 0x00],
+            group: "[Library]",
+            unshift: function() {
+                this.inKey = "MoveVertical";
+            },
+            shift: function() {
+                this.inKey = "MoveFocus";
+            }
+        })
+    });
+
+    const shiftableComponents = this.components;
+
+    this.shift = new components.Button({
+        midi: [0x9E, 0x00],
+        input: function(channel, control, value, status, _group) {
+            shiftableComponents.forEachComponent(comp => comp.disconnect());
+            if (this.isPress(channel, control, value, status)) {
+                shiftableComponents.shift();
+            } else {
+                shiftableComponents.unshift();
+            }
+            shiftableComponents.forEachComponent(comp => {
+                comp.connect();
+                comp.trigger();
+            });
+        },
+    });
+
+    // request controls status per standard serato sysex
+    midi.sendSysexMsg([0xF0, 0x00, 0x20, 0x7F, 0x00, 0xF7]);
+    // request firmware version
+    midi.sendSysexMsg(ReloopReady.requestFirmwareVersionSysex.concat([0xF7]));
+};
+
+ReloopReady.shutdown = () => {
+    ReloopReady.components.shutdown();
+};
+
+
+ReloopReady.Channel = function(index) {
+    const channel = index + 1;
+    const group = `[Channel${channel}]`;
+    const eqGroup = `[EqualizerRack1_${group}_Effect1]`;
+
+    if (ReloopReady.threeBandEQ) {
+        this.knob1 = new components.Pot({
+            midi: [0xB0 + index, 0x16],
+            group: eqGroup,
+            inKey: "parameter3"
+        });
+        this.knob2 = new components.Pot({
+            midi: [0xB0 + index, 0x17],
+            group: eqGroup,
+            inKey: "parameter2"
+        });
+    } else {
+        this.knob1 = new components.Pot({
+            midi: [0xB0 + index, 0x16],
+            inKey: "pregain"
+        });
+
+        this.knob2 = new components.Pot({
+            midi: [0xB0 + index, 0x17],
+            group: eqGroup,
+            inKey: "parameter3"
+        });
+    }
+
+    this.knob3 = new components.Pot({
+        midi: [0xB0 + index, 0x19],
+        group: eqGroup,
+        inKey: "parameter1"
+    });
+
+
+    this.filter = new components.Pot({
+        midi: [0xB0 + index, 0x1A],
+        group: `[QuickEffectRack1_${group}]`,
+        inKey: "super1"
+    });
+
+    this.rate = new components.Pot({
+        midi: [0xB0 + index, 0x09], // MSB control: 0x3F
+        inKey: "rate",
+        // tempo fader response: top ("-") @ 0, bottom ("+") @ 1023
+        max: 1023,
+        invert: true,
+        // see explanation of ReloopReady.tempoFaderMiddles for info
+        inValueScale: function(value) {
+            const middle = (this.max + 1) * ReloopReady.tempoFaderMiddles[index];
+            return script.absoluteNonLinInverse(value, 0, middle, this.max, 0, 1);
+        }
+    });
+
+    // TODO this seems to have soft-takeover enabled sometimes?!
```

Comment:
Is this still a ToDo?

Reply:
not sure, I'll investigate

Reply:
okay, it definitely is. I'm not sure why, whether its a mixxx, or a hardware issue and its not deterministically reproducible for me either.